src/Lecture3.hs

```diff
@@ -178,7 +209,8 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
+appendDiff3 :: (Monoid a, Eq a) => a -> a -> a -> a
+appendDiff3 a b c = mconcat $ nub [a, b, c]
```

Comment:
That's a clever and concise implementation! 👏🏻 

However, the type signature is more strict than required. Your function works with `Monoid`. But not every type can have a `Monoid` instance. This means, that `appendDiff3` can't work with types like `List1`. But if you change the constraint from `Monoid` to `Semigroup`, this function will now work. Though, relaxing the constraint will require changing the implementation as well 🚧

Reply:
So something like this?
```haskell
appendDiff3 :: (Semigroup a, Eq a) => a -> a -> a -> a
appendDiff3 a b c = foldr1 (<>) $ nub [a, b, c]
```

Reply:
That's one possible solution 👍🏻

Reply:
Sounds like you've had a different expected/right solution :)

Btw: I've been thinking about a way to solve it without using nub, but without directly comparing all of the arguments also, and thought about the state monad. Would this function be a good one for learning/understanding the state monad? With something like {last appended :: a, setOfAllAppended :: Set a} as a state?

Reply:
I was thinking about comparing relevant arguments with something like guards and appending only different elements 🙂 
This is good enough for for three arguments 👍🏻 

If you want to generalise this function to something like "list of elements" then an approach with a Set of already appended values works great 🔝  You don't need to use State monad for this though, it'll be a simple recursive function.

Reply:
Yeah, I know that State monad is unnecessary in this case. It's just I've been trying to understand State monad and looking for a simplest case where can I apply it. Almost all the examples in "tutorials" are somehow complicated in one way or another. 
Anyway, I think you're right, and it'd be better to find a case where State monad is more appropriate. Thanks 👍